commit b363b3304bcf68c4541683b2eff70b29f0446a5b
Author: Steve French <sfrench@us.ibm.com>
Date:   Wed Mar 18 05:57:22 2009 +0000

    [CIFS] Fix memory overwrite when saving nativeFileSystem field during mount
    
    CIFS can allocate a few bytes to little for the nativeFileSystem field
    during tree connect response processing during mount.  This can result
    in a "Redzone overwritten" message to be logged.
    
    Signed-off-by: Sridhar Vinay <vinaysridhar@in.ibm.com>
    Acked-by: Shirish Pargaonkar <shirishp@us.ibm.com>
    CC: Stable <stable@kernel.org>
    Signed-off-by: Steve French <sfrench@us.ibm.com>

Backported to Debian's 2.6.18 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.18.orig/fs/cifs/connect.c linux-source-2.6.18/fs/cifs/connect.c
--- linux-source-2.6.18.orig/fs/cifs/connect.c	2008-12-25 14:04:13.000000000 -0700
+++ linux-source-2.6.18/fs/cifs/connect.c	2009-04-29 01:20:52.000000000 -0600
@@ -3151,7 +3151,7 @@ CIFSTCon(unsigned int xid, struct cifsSe
 			    BCC(smb_buffer_response)) {
 				kfree(tcon->nativeFileSystem);
 				tcon->nativeFileSystem =
-				    kzalloc(length + 2, GFP_KERNEL);
+				    kzalloc(2*(length + 1), GFP_KERNEL);
 				cifs_strfromUCS_le(tcon->nativeFileSystem,
 						   (__le16 *) bcc_ptr,
 						   length, nls_codepage);
